# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jsapi-tests/testGCAllocator.cpp
# Commit: 1055931aad82
# Full Hash: 1055931aad820878188250e89d5dacb35b893400
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2025-01-17 21:57:54
# Description:
#   Bug 1941728 - Increase the granularity of large buffer allocations to 1MB r=sfink
#   
#   Currently the granularity is PageSize, but that may be less then the system
#   page size.
#   
# ==============================================================================

diff -r f10cdccac853 -r 1055931aad82 js/src/jsapi-tests/testGCAllocator.cpp
--- a/js/src/jsapi-tests/testGCAllocator.cpp	Fri Jan 17 18:39:40 2025 +0200
+++ b/js/src/jsapi-tests/testGCAllocator.cpp	Fri Jan 17 15:23:05 2025 +0000
@@ -578,8 +578,10 @@
       holder->setBuffer(alloc);
 
       // Realloc to same size.
+      void* prev = alloc;
       alloc = ReallocBuffer(zone, alloc, requestSize, nurseryOwned);
       CHECK(alloc);
+      CHECK(alloc == prev);
       CHECK(actualSize == GetAllocSize(alloc));
       CHECK(IsNurseryOwned(alloc) == nurseryOwned);
       CHECK(CheckAllocData(alloc, actualSize));
